NOTICE

```diff
@@ -1,6 +1,2551 @@
-Elastic OTel Java Agent
-Copyright 2023-2024 Elasticsearch B.V.
+Elastic OTel Java Distribution
+          Copyright 2023-2024 Elasticsearch B.V.
+
+This project is licensed under the Apache License, Version 2.0 - https://www.apache.org/licenses/LICENSE-2.0
+An embedded copy of the Apache License, Version 2.0 is provided below.
+
+This project depends on ASM which is licensed under the 3-Clause BSD License - https://opensource.org/licenses/BSD-3-Clause
+An embedded copy of the ASM license (3-Clause BSD License) is provided below.
+
+This project depends on okhttp, which contains 'publicsuffixes.gz' that is licensed under the Mozilla Public License, v. 2.0 - https://mozilla.org/MPL/2.0/
+An embedded copy of the publicsuffixes.gz license (Mozilla Public License, v. 2.0) is provided below.
```

Comment:
[for reviewer] the complete notice file is now generated, only the first few lines are manually added in the `build.gradle` file.